File: src/dataset/cpp/difference-of-squares_exercism-pawelo091991/openAI/ChatGPT4_difference_of_squares_v2.cpp
#include "difference_of_squares.h"
#include <cmath>

int squares::square_of_sum(int number) {
    int sum = number * (number + 1) / 2; // Use arithmetic series formula
    return sum * sum;
}

int squares::sum_of_squares(int number) {
    return (number * (number + 1) * (2 * number + 1)) / 6; // Use sum of squares formula
}

int squares::difference(int number) {
    return square_of_sum(number) - sum_of_squares(number);
}

File: src/dataset/typescript/difference-of-squares_exercism-samajammin/openAI/ChatGPT4_difference-of-squares_v2.ts
export default class Squares {
  squareOfSum: number;
  sumOfSquares: number;
  difference: number;

  constructor(num: number) {
    const { squareOfSum, sumOfSquares } = this.calculate(num);
    this.squareOfSum = squareOfSum;
    this.sumOfSquares = sumOfSquares;
    this.difference = squareOfSum - sumOfSquares;
  }

  private calculate(num: number): { squareOfSum: number; sumOfSquares: number } {
    let sum = 0;
    let sumOfSquares = 0;

    for (let i = 1; i <= num; i++) {
      sum += i;
      sumOfSquares += i ** 2;
    }

    return { squareOfSum: sum ** 2, sumOfSquares };
  }
}